### PR TITLE
DCS-1450 implement new-arrival cookie in new arrival flow initial steps

### PR DESCRIPTION
**Pull request by louisenorris**

### CODE REVIEW

File: integration_tests/integration/bookedtoday/arrivals/singleExistingRecordFound.spec.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    cy.task('reset')
+    cy.task('stubSignIn', Role.PRISON_RECEPTION)
+    cy.task('stubPrison', 'MDI')
+    cy.task('stubAuthUser')
+    cy.task('stubUserCaseLoads')
+    cy.task('stubTransfers', { caseLoadId: 'MDI', transfers: [] })
+    cy.task('stubMissingPrisonerImage')
+  })
+
+  it('PER record should contain a full set of correctly formatted move data', () => {
```
Comment (louisenorris) [inline integration_tests/integration/bookedtoday/arrivals/singleExistingRecordFound.spec.ts:23]: Currently working to get this test working, as there is a cookie I think we need to implement the complete journey of that cookie for the test to pass - it's currently not redirecting from the sex page to the imprisonment status page which is confusing me as I hardcoded sex as 'MALE' in the cookie therefore the redirect should happen..

File: integration_tests/integration/bookedtoday/arrivals/checkAnswers.spec.ts [hunk 2]
```diff
@@ -23,50 +32,92 @@ context('Check Answers', () => {
...
     imprisonmentStatusPage.continue().click()
+    imprisonmentStatusPage.hasError('Select a reason for imprisonment')
+    imprisonmentStatusPage.imprisonmentStatusRadioButton('determinate-sentence').click()
+    imprisonmentStatusPage.continue().click()
+
+    const movementReasonPage = Page.verifyOnPage(MovementReasonsPage)
+    movementReasonPage.continue().click()
+    movementReasonPage.hasError('Select the type of determinate sentence')
+    movementReasonPage.movementReasonRadioButton('26').click()
+    movementReasonPage.continue().click()
```
Comment (andrewrlee) [inline integration_tests/integration/bookedtoday/arrivals/checkAnswers.spec.ts:63]: munged in a few other page specific tests into this test

File: integration_tests/integration/bookedtoday/arrivals/checkAnswers.spec.ts [hunk 3]
```diff
@@ -23,50 +32,92 @@ context('Check Answers', () => {
...
 
-    const checkAnswersPage = CheckAnswersPage.goTo(expectedArrival.id)
```
Comment (andrewrlee) [inline integration_tests/integration/bookedtoday/arrivals/checkAnswers.spec.ts:37]: Existing, but we shouldn't really be using `goTo` anywhere apart from the entry of a test

File: integration_tests/integration/bookedtoday/arrivals/checkAnswers.spec.ts [hunk 4]
```diff
@@ -23,50 +32,92 @@ context('Check Answers', () => {
...
+      .should('have.attr', 'href')
+      .then(href => {
+        expect(href).to.equal('https://digital-dev.prison.service.justice.gov.uk')
+      })
+    confirmAddedToRollPage.addAnotherToRoll().click()
+    Page.verifyOnPage(ChoosePrisonerPage)
   })
 
-  it('Should not display prison or pnc numbers if none present', () => {
+  it('Should show matched results when found', () => {
```
Comment (andrewrlee) [inline integration_tests/integration/bookedtoday/arrivals/checkAnswers.spec.ts:100]: there's a load of different options we could test here - this was a bit of an arbitrary choice but had to change existing test as no PNC and no prison number is not a valid combo

File: integration_tests/mockApis/responses/expectedArrivals.ts [hunk 5]
```diff
@@ -5,6 +5,7 @@ export default {
     dateOfBirth = '1970-01-01',
     pncNumber = '01/2345A',
     prisonNumber = 'G0015GF',
+    gender = 'MALE',
```
Comment (andrewrlee) [inline integration_tests/mockApis/responses/expectedArrivals.ts:8]: allow specifying gender but default to same value

File: integration_tests/pages/bookedtoday/arrivals/singleRecordFound.ts [hunk 6]
```diff
@@ -1,15 +1,10 @@
 import Page, { PageElement } from '../../page'
 
-export default class ExistingRecordPage extends Page {
+export default class SingleRecordFoundPage extends Page {
   constructor() {
     super('This person has an existing prisoner record')
   }
 
-  static goTo(id: string): ExistingRecordPage {
```
Comment (andrewrlee) [inline integration_tests/pages/bookedtoday/arrivals/singleRecordFound.ts:8]: should never directly go to this page as cookie won't be set up

File: server/routes/bookedtoday/choosePrisonerController.test.ts [hunk 7]
```diff
@@ -229,14 +229,7 @@ describe('GET /confirm-arrival/choose-prisoner/:id', () => {
...
         .expect(res => {
-          expectSettingCookie(res, 'new-arrival').toStrictEqual({
-            firstName: 'Harry',
-            lastName: 'Stanton',
-            dateOfBirth: '1961-01-01',
-            sex: 'MALE',
-            prisonNumber: 'A1234BC',
-            pncNumber: '01/123456',
-          })
+          expectSettingCookie(res, 'new-arrival').toStrictEqual(undefined)
```
Comment (louisenorris) [inline server/routes/bookedtoday/choosePrisonerController.test.ts:232]: shouldn't we expect the cookie to contain arrival data here if we are setting the cookie on route to record-found?

Comment (andrewrlee) [reply server/routes/bookedtoday/choosePrisonerController.test.ts]: fixed!

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**